### ISSUE TITLE
404 Not found in proxy raw repo when 301/302 redirect from upstream

### ISSUE THREAD

**Issue by alexandrovas** [sonatype/nexus-public#315]
Nexus OSS returns 404 in raw proxy repo when upstream returns 301/302.

Nexus version: OSS 3.54.1-01

### How to reproduce

1. Create repo in Nexus with type `raw (proxy)` and name `cypress`
1. Setup remote storage URL which returns 301/302 redirect: `https://cy-cdn.currents.dev`
1. Disable Not found cache in repo
1. Send request to Nexus `<nexus_base_url>/repository/cypress/desktop/9.7.0?platform=linux&arch=x64`), see 404 error
1. Send request to upstream `https://cy-cdn.currents.dev/desktop/9.7.0?platform=linux&arch=x64`, see 302 redirect to `https://cy.pw.run/linux-x64_9.7.0.zip`

### Expected behavior

Nexus should follow redirect and download file from upstream.

**Comment by mrprescott** [comment]
It sounds like you're using a raw repo as a stopgap implementation of a proxy of some more specific ecosystem, is that right? It looks like raw repositories simply don't have the capability to follow redirect in this manner, that logic is a bit simpler than some of the other formats. I'm switching this to an enhancement request so we can measure if other users need this capability as well.

**Comment by alexandrovas** [comment]
@mrprescott yes, you right. I use Nexus raw repo in my company to create internal cache for builds without Internet access.
